src/dao.ts

```diff
@@ -122,6 +123,20 @@ export class DAO implements IStateful<IDAOState> {
     return this.context.getBalance(this.address)
   }
 
+  public approveForStaking(amount: number) {
```

Comment:
doesn't really make sense for this to be in dao.ts since approving GEN for staking is not specific to a DAO

Reply:
good point